edx_analytics_transformers/django/models.py

```diff
@@ -0,0 +1,184 @@
+        verbose_name='Is Enabled'
+    )
+
+    configurations = JSONField()
+
+    class Meta:
+        verbose_name = 'Router Configurations'
+        verbose_name_plural = 'Router Configurations'
+        ordering = ('backend_name', 'is_enabled', '-modified')
+
+    def __str__(self):
+        return '{id} - {backend} - {is_enabled}'.format(
+            id=self.pk,
+            backend=self.backend_name,
+            is_enabled='Enabled' if self.is_enabled else 'Disabled'
+        )
+
+    @classmethod
+    def get_latest_enabled_router(cls, backend_name):
+        """
+        Wrapper method for `_get_latest_cached_router`.
+
+        Return the last modified, enabled router for the backend matching the `backend_name`.
+        Return `None` if there is no filter matching the criteria.
+
+        Arguments:
+            backend_name (str):    Name of the backend for which the filter is required.
+
+        Returns:
+            RouterConfigurations or None
+        """
+        return cls._get_latest_cached_router(backend_name=backend_name)
+
+    @classmethod
+    def _get_latest_cached_router(cls, backend_name):
+        """
+        Return the last modified, enabled router for the backend matching the `backend_name`.
+
+        First look for the router in the cache and return its value from there if found.
+        If not found in the cache, call the `_get_latest_enabled_router` method to get the
+        router and store it in the cache before returning it.
+
+        Arguments:
+            backend_name (str):    Name of the backend for which the router is required.
+
+        Returns:
+            RouterConfigurations` or None
+        """
+        router_cache_key = get_cache_key(namespace=ROUTER_CACHE_NAMESPACE, backend_name=backend_name)
+        cache_response = TieredCache.get_cached_response(router_cache_key)
+
+        if cache_response.is_found:
+            logger.info('Router is found in cache for backend "%s"', backend_name)
+            router = cache_response.value
+        else:
+            logger.info('No router was found in cache for backend "%s"', backend_name)
+            router = cls._get_latest_enabled_router(backend_name=backend_name)
+
+            TieredCache.set_all_tiers(router_cache_key, router)
+            logger.info('Router has been stored in cache for backend "%s"', backend_name)
+
+        return router
+
+    @classmethod
+    def _get_latest_enabled_router(cls, backend_name):
+        """
+        Return the last modified, enabled router for the backend matching the `backend_name`.
+
+        Return `None` if there is no filter matching the criteria.
+
+        Arguments:
+            backend_name (str):    Name of the backend for which the filter is required.
+
+        Returns:
+            RouterConfigurations or None
+        """
+        queryset = cls.objects.filter(backend_name=backend_name, is_enabled=True)
+
+        return queryset.latest('modified') if queryset.exists() else None
```

Comment:
Yes `latest` will raise the exception in case but `queryset.exists()` check takes care of that and returns `None` if queryset is empty. `filter(..).last()` on other hand will raise the exception too and we will have to handle it. I think it won't make any difference.

Reply:
@ziafazal to use `last` we have to do it like this:
```
cls.objects.filter(backend_name=backend_name, is_enabled=True).order_by('modified').last()
```

because `last` simple returns the last object from the queryset and doesn't do any ordering by itself. We have to do it like this.

We have `ordering` in the model's meta class so we don't need the `order_by` clause here. but we will be dependent on the ordering then. to make it failure proof and independent, we have to add the ordering here.